File: Python/7.4.py
# 7.4. Write a Python function that prints out the first n rows of Pascal's triangle.

def pascalsTriangle(n):
    triangle=[[1]]
    for i in range(1,n):
        row=[1]
        for j in range(1, i):
            row.append(triangle[i-1][j-1] + triangle[i-1][j])
        row.append(1)
        triangle.append(row)
    return triangle
n=int(input("Enter the number of rows for Pascal's triangle: "))
for row in pascalsTriangle(n):
    print(row)

File: C/7.4.c
/* 
7.4. Write a program in C to find the factorial of a number using recursion.
*/

#include <stdio.h>
int factorial(int n) {
    if (n == 0 || n == 1)
        return 1;
    else
        return n * factorial(n - 1);
}
int main() {
    int num;
    printf("Enter a number: ");
    scanf("%d", &num);
    printf("Factorial of %d is %d\n", num, factorial(num));
    return 0;
}
